### PR TITLE
validating playbook deps

### PR DESCRIPTION
**Pull request by jochman**
## Status
Ready

fixes: https://github.com/demisto/etc/issues/21394

## Must have
- [X] Tests

### CODE REVIEW

File: Tests/scripts/infrastructure_tests/update_id_set_test.py [hunk 1]
```diff
@@ -161,5 +162,33 @@ def test_get_playbook_data(self):
...
+     False),
+    ({}, False),
+    ({"playbooks": [{"str": PLAYBOOK_DATA}],
+      "TestPlaybooks": [{"str": PLAYBOOK_DATA2}, {"str1": PLAYBOOK_DATA3}]}, False),
+    ({"playbooks": [{"str": PLAYBOOK_DATA}], "TestPlaybooks": [{"str": PLAYBOOK_DATA2}]}, True)
+]
+
+
+@pytest.mark.parametrize('input_data, expected_error', playbook_deps)
+def test_validate_playbook_deps(input_data, expected_error):
```
Comment (anara123) [inline Tests/scripts/infrastructure_tests/update_id_set_test.py]: this test is not clear at all. please rewrite it, for each behaviour write a separate unit test. in unit test explain the behaviour/use case.
What was in the system? what was the trigger? then what you expect to happen?

scenario: playbook depend on sub playbook that don't exists
given
- playbook A depend on sub playbook B

when
- running validate

then
- validate should raise an KeyError exception ...

Like that

**Comment by yaakovi** [comment]
just thought about another issue here:
playbook1 has fromversion 5.5
playbook2 uses playbook1 and has fromversion 4.0
this wont fail the current check

**Comment by jochman** [comment]
> just thought about another issue here:
> playbook1 has fromversion 5.5
> playbook2 uses playbook1 and has fromversion 4.0
> this wont fail the current check

kets merge this and open an issue for dev-task enhancement.

**Comment by Itay4** [comment]
@jochman whats our status here?

**Comment by jochman** [comment]
Rdy

**Comment by Itay4** [comment]
@yaakovi please review

**Pull request closed without merging**